c7n/resources/glacier.py

```diff
@@ -93,3 +96,70 @@ def _augment(r):
 
         return super(GlacierCrossAccountAccessFilter, self).process(
             resources, event)
+
+
+@Glacier.action_registry.register('remove-statements')
+class RemovePolicyStatement(RemovePolicyBase):
+    """Action to remove policy statements from Glacier
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: glacier-cross-account
+                resource: glaier
+                filters:
+                  - type: cross-account
+                actions:
+                  - type: remove-statements
+                    statement_ids: matched
+    """
+
+    permissions = ('glacier:SetVaultAccessPolicy', 'glacier:GetVaultAccessPolicy')
+
+    def process(self, resources):
+        results = []
+        client = local_session(self.manager.session_factory).client('glacier')
+        for r in resources:
+            try:
+                if self.process_resource(client, r):
+                    results.append(r)
```

Comment:
the results to append and return here is the return value from process_resource, the resources are already recorded from the policy.